sklearn/tree/_criterion.pyx

```diff
@@ -347,8 +347,9 @@ cdef class ClassificationCriterion(Criterion):
 
         # Reset to pos=start
         self.reset()
+        return 0
```

Comment:
I'm not sure you're right:
> ```
> cdef int spam() except -1:
>    ...
> ```
> With this declaration, whenever an exception occurs inside spam, it will immediately return with the value -1. Furthermore, whenever a call to spam returns -1, an exception will be assumed to have occurred and will be propagated.

From [here](http://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#error-return-values)